controls/views.py

```diff
@@ -542,6 +544,28 @@ def system_element(request, system_id, element_id):
         }
         return render(request, "systems/element_detail_tabs.html", context)
 
+@login_required
+def new_element(request):
+    """Form to create new system element (aka component)"""
+
+    if request.method == 'POST':
+      form = ElementForm(request.POST)
+      if form.is_valid():
```

Comment:
Might need to have an "else" statement to pair here, in case the form is invalid, to display an error message to the user and re-render the form page or redirect to the Component Library.

Reply:
I added an else on 553 hope this covers it.

Reply:
The `else` block on 553 is not necessary as Django forms handles the routing back to the form page with error messages. So removing.